src/main/java/plugily/projects/buildbattle/commands/TabCompletion.java

```diff
@@ -1,66 +1,107 @@
-
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandSender;
 import org.bukkit.command.TabCompleter;
 import org.bukkit.entity.Player;
+import org.bukkit.util.StringUtil;
+import org.jetbrains.annotations.NotNull;
 
-import plugily.projects.buildbattle.arena.ArenaRegistry;
-import plugily.projects.buildbattle.arena.impl.BaseArena;
-import plugily.projects.buildbattle.commands.arguments.ArgumentsRegistry;
-import plugily.projects.buildbattle.commands.arguments.data.CommandArgument;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 
-/**
- * @author Plajer
- * <p>
- * Created at 27.05.2018
- */
 public class TabCompletion implements TabCompleter {
+    private final HashMultimap<String, String> completesListMap;
 
-  private ArgumentsRegistry registry;
-
-  public TabCompletion(ArgumentsRegistry registry) {
-    this.registry = registry;
-  }
+    public TabCompletion() {
+        completesListMap = HashMultimap.create();
+        add("bb", "join", "randomjoin", "stats", "leave", "top", "create");
+        add("bba", "list", "stop", "forcestart", "reload", "delete", "addplot", "removeplot", "addnpc", "settheme", "votes", "plotwand");
+        add("arena", "[arena]");
+        add("soloteam", "[solo/team]");
+        add("onlineplayer", "(online player)");
+        add("statistic", "[statistic]");
+        add("theme", "(theme)");
+        add("plotID", "[plot ID]");
+        add("addset", "[add/set]");
+        add("amount", "[amount]");
+        add("player", "(player)");
+    }
 
-  @Override
-  public List<String> onTabComplete(CommandSender sender, Command cmd, String label, String[] args) {
-    if (!(sender instanceof Player)) {
-      return Collections.emptyList();
+    private void add(String key, String... args) {
+        completesListMap.putAll(key, Arrays.asList(args));
     }
-    if (cmd.getName().equalsIgnoreCase("buildbattleadmin") && args.length == 1) {
-      return registry.getMappedArguments().get(cmd.getName().toLowerCase()).stream().map(CommandArgument::getArgumentName).collect(Collectors.toList());
+
+    private List<String> getPartial(String token, Iterable<String> collection) {
+        return StringUtil.copyPartialMatches(token, collection, new ArrayList<>());
     }
-    if (cmd.getName().equalsIgnoreCase("buildbattle")) {
-      if (args.length == 2 && args[0].equalsIgnoreCase("join")) {
-        return ArenaRegistry.getArenas().stream().map(BaseArena::getID).collect(Collectors.toList());
-      }
-      if (args.length == 1) {
-        return registry.getMappedArguments().get(cmd.getName().toLowerCase()).stream().map(CommandArgument::getArgumentName).collect(Collectors.toList());
-      }
+
+    public
+    @Override List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command cmd, @NotNull String label, String[] args) {
```

Comment:
```suggestion
    @Override
    public List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command cmd, @NotNull String label, String[] args) {
```

Reply:
sorry, idk what the suggested change is :/

Reply:
`public @Override` is an IDE suggested bullshit thing, use
```java
@Override
public
```